site/src/sphinx/client-decorator.rst

```diff
@@ -67,6 +67,116 @@ If your decorator is expected to be reusable, it is recommended to define a new
     // Using a lambda expression:
     cb.decorator(delegate -> new AuditClient(delegate));
 
+The order of decoration
+-----------------------
+
+The decorators are executed in reverse order of the insertion. The following example shows which order
+the decorators are executed by printing the messages.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.WebClient;
+
+    ClientBuilder cb = Clients.builder(...);
+
+    // #2 decorator
+    cb.decorator((delegate, ctx, req) -> {
+        System.err.println("Secondly, executed.");
+        ...
+    });
+
+    // #1 decorator.
+    // No matter decorator() or option() is used, decorators are executed in reverse order of the insertion.
+    cb.option(ClientOption.DECORATION, ClientDecoration.of((delegate, ctx, req) -> {
+        System.err.println("Firstly, executed");
+        ...
+    });
+
+    WebClient myClient = cb.build(WebClient.class);
+
+The following diagram describes how an HTTP request and HTTP response are gone through decorators:
+
+.. uml::
+
+    @startditaa
+    +-----------+  req  +-----------+  req  +-----------+  req  +------------------+    req    +--------------+
+    |           |------>|           |------>|           |------>|                  |---------->|              |
+    | WebClient |       | #1        |       | #2        |       | Armeria          |           |    Server    |
+    |           |  res  | decorator |  res  | decorator |  res  | Networking Layer |    res    |              |
+    |           |<------|           |<------|           |<------|                  |<----------|              |
+    +-----------+       +-----------+       +-----------+       +------------------+           +--------------+
+    @endditaa
+
+If the client is a Thrift client and RPC decorators are inserted, HTTP decorators and RPC decorators are
+separately grouped and executed in reverse order of the insertion:
+
+.. code-block:: java
+
+    ClientBuilder cb = Clients.builder(...);
+
+    // #2 HTTP decorator.
+    cb.decorator((delegate, ctx, httpReq) -> {
+        System.err.println("Fourthly, executed.");
+        ...
+    });
+
+    // #2 RPC decorator.
+    cb.rpcDecorator((delegate, ctx, rpcReq) -> {
+        System.err.println("Secondly, executed.");
+        ...
+    });
+
+    // #1 HTTP decorator.
+    cb.decorator((delegate, ctx, httpReq) -> {
+        System.err.println("Thirdly, executed.");
+        ...
+    });
+
+    // #1 RPC decorator.
+    cb.rpcDecorator((delegate, ctx, rpcReq) -> {
+        System.err.println("Firstly, executed.");
+        ...
+    });
+
+An RPC request is converted into an HTTP request before it's sent to a server.
+Therefore, RPC decorators are inserted before the RPC request is converted and HTTP decorators are inserted
```

Comment:
How about using `apply` instead of `insert`
For example:
```suggestion
Therefore, RPC decorators are applied before the RPC request is converted and HTTP decorators are applied
```

Reply:
That's a good suggestion. Fixed. 😉